src/BitmovinPlayer.tsx

```diff
@@ -201,12 +201,20 @@ function destroyPlayer(
   player: PlayerAPI,
   rootContainerElement: HTMLDivElement,
   playerContainerElement: HTMLDivElement,
+  uiManager: UIManager | undefined,
 ) {
   playerContainerElement.style.display = 'none';
 
   const removePlayerContainerElement = () => {
     rootContainerElement.removeChild(playerContainerElement);
   };
 
+  // First destroy the UI manager if it exists
+  if (uiManager) {
+    uiManager.release();
+    uiManager = undefined;
```

Comment:
@ReallyBadNews, this line (215) is unnecessary and should be removed.

I think this is the last remark. 🙂